app/app/src/main/kotlin/com/hedvig/android/app/ui/HedvigAppState.kt

```diff
@@ -187,27 +200,30 @@ private fun TopLevelDestinationNavigationSideEffect(
   DisposableEffect(navController, tabNotificationBadgeService, coroutineScope) {
```

Comment:
Why is it better to have a DisposableEffect here and not LaunchEffect if we have to pass a coroutineScope as a key here anyway, and it's default rememberCoroutineScope()? Can't find where it might be changed

Reply:
It's because what we do inside here is we register a listener to the navController.
So we need to make 100% sure that whatever we regiester, we also _unregister_ once this leaves composition, or one of the keys change (and we then add the new listener in that case)
This is done here with the onDispose https://github.com/HedvigInsurance/android/blob/14eaa9cbb8cbe59d074790c62e754b5fc7738460/app/app/src/main/kotlin/com/hedvig/android/app/ui/HedvigAppState.kt#L239-L241 which DispoableEffect forces you to call in the end.

Read more here https://developer.android.com/jetpack/compose/side-effects#disposableeffect 😊